spring-batch-docs/asciidoc/readersAndWriters.adoc

```diff
@@ -2779,221 +2779,255 @@ reconstituted from the execution context if the stream was
+===== `MongoItemWriter`
 The `MongoItemWriter` is an `ItemWriter` implementation that writes to a
 MongoDB store using an implementation of Spring Data's `MongoOperations`.
 Spring Batch provides a `MongoItemWriterBuilder` to construct an instance
 of the `MongoItemWriter`.
 
 [[repositoryItemWriter]]
-===== RepositoryItemWriter
+===== `RepositoryItemWriter`
 The `RepositoryItemWriter` is an `ItemWriter` wrapper for a `CrudRepository` from Spring Data.
 Spring Batch provides a `RepositoryItemWriterBuilder` to construct an instance
 of the `RepositoryItemWriter`.
 
 [[hibernateItemWriter]]
-===== HibernateItemWriter
+===== `HibernateItemWriter`
 The `HibernateItemWriter` is an `ItemWriter` that uses a Hibernate session to save or update entities
 that are not part of the current Hibernate session.
 Spring Batch provides a `HibernateItemWriterBuilder` to construct an instance
 of the `HibernateItemWriter`.
 
 [[jdbcBatchItemWriter]]
-===== JdbcBatchItemWriter
+===== `JdbcBatchItemWriter`
 The `JdbcBatchItemWriter` is an `ItemWriter` that uses the batching features from
 `NamedParameterJdbcTemplate` to execute a batch of statements for all items
 provided.
 Spring Batch provides a `JdbcBatchItemWriterBuilder` to construct an instance
 of the `JdbcBatchItemWriter`.
 
 [[jpaItemWriter]]
-===== JpaItemWriter
-The `JpaItemWriter` is an `ItemWriter` that is using a JPA EntityManagerFactory
-to merge any Entities that aren't part of the persistence context.
+===== `JpaItemWriter`
+The `JpaItemWriter` is an `ItemWriter` that uses a JPA `EntityManagerFactory`
+to merge any entities that are not part of the persistence context.
 Spring Batch provides a `JpaItemWriterBuilder` to construct an instance
 of the `JpaItemWriter`.
 
 [[specializedReaders]]
 ==== Specialized Readers
-Here are some specialized readers offered by Spring Batch out of the box.
+Spring Batch offers the following specialized readers:
+
+* <<ldifReader>>
+* <<mappingLdifReader>>
 
 [[ldifReader]]
-===== LdifReader
-The `LdifReader` reads LDIF (LDAP Data Interchange Format) records from a `Resource`, parses them
+===== `LdifReader`
+The `LdifReader` reads LDIF (LDAP Data Interchange Format) records from a `Resource`, parses them,
 and returns a `LdapAttribute` object for each `read` executed.
 Spring Batch provides a `LdifReaderBuilder` to construct an instance of the `LdifReader`.
 
 
 [[mappingLdifReader]]
-===== MappingLdifReader
+===== `MappingLdifReader`
 The `MappingLdifReader` reads LDIF (LDAP Data Interchange Format) records from a `Resource`, parses them
 then maps each LDIF record to a POJO (Plain Old Java Object). Each read returns a POJO.
 Spring Batch provides a `MappingLdifReaderBuilder` to construct an instance of the `MappingLdifReader`.
 
 [[specializedWriters]]
 ==== Specialized Writers
-Here are some specialized writers offered by Spring Batch out of the box.
+Spring Batch offers the following specialized writers:
+
+* <<classifierCompositeItemWriter>>
```

Comment:
ClassifierCompositeItemWriter is really a decorator and should be moved to that section...

Reply:
OK. You might want to kick this back to me, then. That's a link to an anchor later in the document. Consequently, the link, the anchor, and the heading all have to get moved.